--- chat.h.orig	2025-03-13 13:05:46
+++ chat.h	2025-03-13 13:02:13
@@ -8,8 +8,17 @@
 #include <string>
 #include <vector>
 #include <map>
+#include "minja/chat-template.hpp"
+#include "minja/minja.hpp"
+
+typedef minja::chat_template common_chat_template;
+
+struct common_chat_templates {
+    bool has_explicit_template; // Model had builtin template or template overridde was specified.
+    std::unique_ptr<common_chat_template> template_default; // always set (defaults to chatml)
+    std::unique_ptr<common_chat_template> template_tool_use;
+};
 
-struct common_chat_templates;
 
 struct common_chat_tool_call {
     std::string name;
