--- chat.cpp.orig	2025-08-26 19:14:43
+++ chat.cpp	2025-08-26 19:06:49
@@ -6,9 +6,6 @@
 #include "log.h"
 #include "regex-partial.h"

-#include <minja/chat-template.hpp>
-#include <minja/minja.hpp>
-
 #include <cstdio>
 #include <exception>
 #include <iostream>
@@ -122,16 +119,6 @@
     }
     return diffs;
 }
-
-typedef minja::chat_template common_chat_template;
-
-struct common_chat_templates {
-    bool add_bos;
-    bool add_eos;
-    bool has_explicit_template; // Model had builtin template or template overridde was specified.
-    std::unique_ptr<common_chat_template> template_default; // always set (defaults to chatml)
-    std::unique_ptr<common_chat_template> template_tool_use;
-};

 struct templates_params {
     json messages;
@@ -784,8 +771,7 @@
     if (additional_context) {
         tmpl_inputs.extra_context.merge_patch(*additional_context);
     }
-    // TODO: add flag to control date/time, if only for testing purposes.
-    // tmpl_inputs.now = std::chrono::system_clock::now();
+    tmpl_inputs.now = inputs.now;

     minja::chat_template_options tmpl_opts;
     // To avoid double BOS / EOS tokens, we're manually removing begining / trailing tokens
