--- chat.cpp.orig	2025-03-13 13:05:47
+++ chat.cpp	2025-03-13 13:02:10
@@ -1,18 +1,8 @@
 #include "chat.h"
 #include "json-schema-to-grammar.h"
 #include "log.h"
-#include "minja/chat-template.hpp"
-#include "minja/minja.hpp"
 
 #include <optional>
-
-typedef minja::chat_template common_chat_template;
-
-struct common_chat_templates {
-    bool has_explicit_template; // Model had builtin template or template overridde was specified.
-    std::unique_ptr<common_chat_template> template_default; // always set (defaults to chatml)
-    std::unique_ptr<common_chat_template> template_tool_use;
-};
 
 struct templates_params {
     json messages;
