--- common.h.orig	2025-03-13 13:05:47
+++ common.h	2025-03-13 13:04:21
@@ -224,6 +224,7 @@ enum common_reasoning_format {
 };
 
 struct common_params {
+    bool vocab_only               = false;
     int32_t n_predict             =    -1; // new tokens to predict
     int32_t n_ctx                 =  4096; // context size
     int32_t n_batch               =  2048; // logical batch size for prompt processing (must be >=32 to use BLAS)
@@ -343,6 +344,9 @@ struct common_params {
 
     bool single_turn       = false; // single turn chat conversation
 
+    llama_progress_callback progress_callback = nullptr;
+    void * progress_callback_user_data = nullptr;
+
     lm_ggml_type cache_type_k = LM_GGML_TYPE_F16; // KV cache data type for the K
     lm_ggml_type cache_type_v = LM_GGML_TYPE_F16; // KV cache data type for the V