--- ggml-metal/kernels/mul_mv.metal.orig
+++ ggml-metal/kernels/mul_mv.metal
@@ -3096,7 +3096,7 @@
         device       char * dst,
         threadgroup  char * shmem,
         uint3  tgpig,
-        ushort tiitg,
+        uint   tiitg,
         ushort tiisg,
         ushort sgitg) {
     disp_fn(args, src0, src1, dst, tgpig, tiisg);
@@ -3110,7 +3110,7 @@
         device       char * dst,
         threadgroup  char * shmem,
         uint3  tgpig,
-        ushort tiitg,
+        uint   tiitg,
         ushort tiisg,
         ushort sgitg) {
     disp_fn(args, src0, src1, dst, shmem, tgpig, tiisg, sgitg);
@@ -3127,7 +3127,7 @@
         device const char * ids,
         threadgroup  char * shmem [[threadgroup(0)]],
         uint3  tgpig[[threadgroup_position_in_grid]],
-        ushort tiitg[[thread_index_in_threadgroup]],
+        uint   tiitg[[thread_index_in_threadgroup]],
         ushort tiisg[[thread_index_in_simdgroup]],
         ushort sgitg[[simdgroup_index_in_threadgroup]]) {
     const int iid1 = tgpig.z/args.nei0;
