--- ggml-metal-device.m.orig	2025-11-10 17:26:40
+++ ggml-metal-device.m	2025-11-10 17:22:40
@@ -181,7 +181,15 @@
         NSBundle * bundle = [NSBundle bundleForClass:[LMGGMLMetalClass class]];
 #endif

-        NSString * path_lib = [bundle pathForResource:@"default" ofType:@"metallib"];
+#if TARGET_OS_SIMULATOR
+        NSString *lib_name = @"ggml-llama-sim";
+        if (dev->props.has_tensor) lib_name = @"ggml-llama-sim-metal4";
+        NSString * path_lib = [bundle pathForResource:@"ggml-llama-sim" ofType:@"metallib"];
+#else
+        NSString *lib_name = @"ggml-llama";
+        if (dev->props.has_tensor) lib_name = @"ggml-llama-metal4";
+        NSString * path_lib = [bundle pathForResource:@"ggml-llama" ofType:@"metallib"];
+#endif
         if (path_lib == nil) {
             // Try to find the resource in the directory where the current binary located.
             NSString * bin_cur = [[NSProcessInfo processInfo] arguments][0];
