--- ggml-metal-device.cpp.orig	2025-09-18 11:33:02
+++ ggml-metal-device.cpp	2025-09-18 11:26:49
@@ -72,7 +72,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -89,7 +91,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -116,7 +120,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -133,7 +139,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -150,7 +158,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -167,7 +177,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -225,7 +237,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -259,7 +273,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -288,7 +304,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -317,7 +335,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -342,7 +362,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -363,7 +385,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -405,7 +429,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -422,7 +448,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -439,7 +467,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 8192);
 
@@ -623,7 +653,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_nr0 (res, nr0);
     lm_ggml_metal_pipeline_set_nr1 (res, nr1);
@@ -645,7 +677,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     const size_t smem = (size_t) ne02*ne20*sizeof(uint16_t);
 
@@ -666,7 +700,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 8192);
 
@@ -832,7 +868,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_nr0 (res, nr0);
     lm_ggml_metal_pipeline_set_nr1 (res, nr1);
@@ -858,7 +896,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*(sizeof(float) + sizeof(int32_t)));
 
@@ -888,7 +928,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1079,7 +1121,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1107,7 +1151,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -1131,7 +1177,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -1154,7 +1202,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -1178,7 +1228,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     lm_ggml_metal_pipeline_set_smem(res, 32*sizeof(float));
 
@@ -1216,7 +1268,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1239,7 +1293,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1264,7 +1320,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1283,7 +1341,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1302,7 +1362,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1321,7 +1383,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1340,7 +1404,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
@@ -1359,7 +1425,9 @@
         return res;
     }
 
-    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, nullptr);
+    lm_ggml_metal_cv_t cv = lm_ggml_metal_cv_init();
+    res = lm_ggml_metal_library_compile_pipeline(lib, base, name, cv);
+    lm_ggml_metal_cv_free(cv);
 
     return res;
 }
