--- common/jinja/caps.cpp.orig	2025-01-20 14:24:00
+++ common/jinja/caps.cpp	2025-01-20 14:24:00
@@ -4,7 +4,7 @@

 // note: the json dependency is only for defining input in a convenient way
 // we can remove it in the future when we figure out a better way to define inputs using jinja::value
-#include <nlohmann/json.hpp>
+#include "../../nlohmann/json.hpp"

 #include <functional>
 #include <sstream>
