--- common.cpp.orig	2023-12-19 08:18:55
+++ common.cpp	2023-12-19 08:18:26
@@ -41,6 +41,12 @@
 #if defined(_MSC_VER)
 #pragma warning(disable: 4244 4267) // possible loss of data
 #endif
+
+// build info
+int LLAMA_BUILD_NUMBER = 0;
+char const *LLAMA_COMMIT = "unknown";
+char const *LLAMA_COMPILER = "unknown";
+char const *LLAMA_BUILD_TARGET = "unknown";
 
 int32_t get_num_physical_cores() {
 #ifdef __linux__
