--- common.cpp.orig	2024-03-20 12:34:08
+++ common.cpp	2024-03-20 12:34:16
@@ -44,6 +44,12 @@
 #if defined(_MSC_VER)
 #pragma warning(disable: 4244 4267) // possible loss of data
 #endif
+
+// build info
+int LLAMA_BUILD_NUMBER = 0;
+char const *LLAMA_COMMIT = "unknown";
+char const *LLAMA_COMPILER = "unknown";
+char const *LLAMA_BUILD_TARGET = "unknown";

 #if (defined(LM_GGML_USE_CUBLAS) || defined(LM_GGML_USE_SYCL))
 #define LM_GGML_USE_CUBLAS_SYCL
