--- ggml-backend-reg.cpp.orig	2025-01-20 20:12:46
+++ ggml-backend-reg.cpp	2025-01-20 20:12:48
@@ -35,9 +35,14 @@
 #endif
 
 #ifdef LM_GGML_USE_METAL
+#include <TargetConditionals.h>
+
+#if !TARGET_OS_SIMULATOR
 #include "ggml-metal.h"
 #endif
 
+#endif
+
 #ifdef LM_GGML_USE_SYCL
 #include "ggml-sycl.h"
 #endif
@@ -157,8 +162,12 @@ struct lm_ggml_backend_registry {
         register_backend(lm_ggml_backend_cuda_reg());
 #endif
 #ifdef LM_GGML_USE_METAL
+
+#if !TARGET_OS_SIMULATOR
         register_backend(lm_ggml_backend_metal_reg());
 #endif
+
+#endif
 #ifdef LM_GGML_USE_SYCL
         register_backend(lm_ggml_backend_sycl_reg());
 #endif
