--- ggml-metal.m.orig	2023-10-27 12:54:53
+++ ggml-metal.m	2023-10-27 12:54:54
@@ -209,7 +209,7 @@
         } else {
             LM_GGML_METAL_LOG_INFO("%s: default.metallib not found, loading from source\n", __func__);
 
-            NSString * sourcePath = [bundle pathForResource:@"ggml-metal" ofType:@"metal"];
+            NSString * sourcePath = [bundle pathForResource:@"ggml-metal-llama" ofType:@"metal"];
             LM_GGML_METAL_LOG_INFO("%s: loading '%s'\n", __func__, [sourcePath UTF8String]);
             NSString * src = [NSString stringWithContentsOfFile:sourcePath encoding:NSUTF8StringEncoding error:&error];
             if (error) {
