--- ggml-metal.m.orig	2024-01-19 10:06:53
+++ ggml-metal.m	2024-01-19 10:06:54
@@ -288,7 +288,7 @@
             if (ggmlMetalPathResources) {
                 sourcePath = [ggmlMetalPathResources stringByAppendingPathComponent:@"ggml-metal.metal"];
             } else {
-                sourcePath = [bundle pathForResource:@"ggml-metal" ofType:@"metal"];
+                sourcePath = [bundle pathForResource:@"ggml-metal-llama" ofType:@"metal"];
             }
             if (sourcePath == nil) {
                 LM_GGML_METAL_LOG_WARN("%s: error: could not use bundle path to find ggml-metal.metal, falling back to trying cwd\n", __func__);
