--- ggml-metal.m.orig	2023-12-19 07:48:34
+++ ggml-metal.m	2023-12-19 07:48:35
@@ -265,7 +265,7 @@
             if (ggmlMetalPathResources) {
                 sourcePath = [ggmlMetalPathResources stringByAppendingPathComponent:@"ggml-metal.metal"];
             } else {
-                sourcePath = [bundle pathForResource:@"ggml-metal" ofType:@"metal"];
+                sourcePath = [bundle pathForResource:@"ggml-metal-llama" ofType:@"metal"];
             }
             if (sourcePath == nil) {
                 LM_GGML_METAL_LOG_WARN("%s: error: could not use bundle path to find ggml-metal.metal, falling back to trying cwd\n", __func__);
