--- ggml-backend.cpp.orig	2024-11-02 18:37:57
+++ ggml-backend.cpp	2024-11-02 18:39:36
@@ -575,8 +575,11 @@
         register_backend(lm_ggml_backend_cuda_reg());
 #endif
 #ifdef LM_GGML_USE_METAL
+#include <TargetConditionals.h>
+#if !TARGET_OS_SIMULATOR
         register_backend(lm_ggml_backend_metal_reg());
 #endif
+#endif
 #ifdef LM_GGML_USE_SYCL
         register_backend(lm_ggml_backend_sycl_reg());
 #endif
