--- ggml-backend-reg.cpp.orig	2024-12-20 13:38:33
+++ ggml-backend-reg.cpp	2024-12-20 13:38:34
@@ -35,9 +35,14 @@
 #endif

 #ifdef LM_GGML_USE_METAL
+#include <TargetConditionals.h>
+
+#if !TARGET_OS_SIMULATOR
 #include "ggml-metal.h"
 #endif

+#endif
+
 #ifdef LM_GGML_USE_SYCL
 #include "ggml-sycl.h"
 #endif
@@ -142,7 +147,11 @@
         register_backend(lm_ggml_backend_cuda_reg());
 #endif
 #ifdef LM_GGML_USE_METAL
+
+#if !TARGET_OS_SIMULATOR
         register_backend(lm_ggml_backend_metal_reg());
+#endif
+
 #endif
 #ifdef LM_GGML_USE_SYCL
         register_backend(lm_ggml_backend_sycl_reg());
